javatests/arcs/core/data/proto/SchemaProtoDecoderTest.kt

```diff
@@ -52,4 +52,150 @@ class SchemaProtoDecoderTest {
         assertThat(schema.fields.singletons).isEqualTo(
             mapOf("text" to FieldType.Text, "bool" to FieldType.Boolean))
     }
+
+    @Test
+    fun decodesCollectionsInSchemaProto() {
+        val schemaProtoText = """
+        fields {
+          key: "text"
+          value: {
+            collection {
+              collection_type: {
+                primitive: TEXT  
+              } 
+            }
+          }
+        }
+        fields {
+          key: "bool"
+          value: {
+            collection {
+              collection_type: {
+                primitive: BOOLEAN
+              } 
+            }
+          }
+        }
+        """.trimIndent()
+        val schema = decodeSchemaProtoText(schemaProtoText)
+        assertThat(schema.fields.collections).isEqualTo(
+            mapOf("text" to FieldType.Text, "bool" to FieldType.Boolean))
+    }
+
+    @Test
+    fun decodesCollectionsOfReferencesInSchemaProto() {
+        val schemaProtoText = """
```

Comment:
Can you also add the Arcs schema as a comment here and else where? It is kind of hard to parse the proto.

Reply:
👍